modules/core/src/main/scala/org/tessellation/infrastructure/snapshot/GlobalSnapshotTraverse.scala

```diff
@@ -48,29 +50,47 @@ object GlobalSnapshotTraverse {
 
               loadInc(lastHash)
                 .onError(_ => logger.error(s"Error during hash chain discovery at ${lastOrdinal.show} with hash ${lastHash.show}"))
-                .map {
-                  case Some(inc) => (hashes.prepend(inc.lastSnapshotHash), inc.ordinal.partialPrevious).asLeft
-                  case None      => (hashes, lastOrdinal).asRight
+                .flatMap {
+                  case Some(inc) =>
+                    loadInfo(inc.ordinal).map {
+                      case Some(_) =>
+                        (hashes, lastOrdinal).asRight[(NonEmptyChain[Hash], Option[SnapshotOrdinal])]
+                      case None =>
+                        (hashes.prepend(inc.lastSnapshotHash), inc.ordinal.partialPrevious)
+                          .asLeft[(NonEmptyChain[Hash], Option[SnapshotOrdinal])]
+                    }
+
+                  case None =>
+                    (hashes, lastOrdinal).asRight[(NonEmptyChain[Hash], Option[SnapshotOrdinal])].pure[F]
                 }
           }.flatMap {
             case (hashes, lastOrdinal) =>
-              val globalHashCandidate = hashes.head
-
-              NonEmptyChain.fromChain(hashes.tail) match {
-                case Some(incHashes) =>
-                  logger
-                    .info(s"Finished rollback chain discovery with global hash candidate $globalHashCandidate at $lastOrdinal")
-                    .as((globalHashCandidate, incHashes))
-                case None => RollbackSnapshotNotFound(rollbackHash).raiseError[F, (Hash, NonEmptyChain[Hash])]
-              }
+              val hashCandidate = hashes.head
+
+              loadInc(hashCandidate)
+                .map(_.fold(hashes.tail)(_ => hashes.toChain))
+                .flatMap(c =>
+                  NonEmptyChain.fromChain(c) match {
+                    case Some(incHashes) =>
+                      logger
+                        .info(s"Finished rollback chain discovery with hash candidate $hashCandidate at $lastOrdinal")
+                        .as((hashCandidate, incHashes))
+                    case None => RollbackSnapshotNotFound(rollbackHash).raiseError[F, (Hash, NonEmptyChain[Hash])]
+                  }
+                )
           }
 
         for {
-          (globalHashCandidate, incHashesNec) <- discoverHashesChain(rollbackHash)
-          global <- loadFullOrErr(globalHashCandidate)
+          (hashCandidate, incHashesNec) <- discoverHashesChain(rollbackHash)
+          _ <- logger.info(s"Rollback hash candidate: ${hashCandidate.show}")
```

Comment:
I think it's a repeated log, there is semantically the same log in `discoverHashesChain` function.